crates/foundry-ui/src/components/typed.rs

```diff
@@ -0,0 +1,40 @@
+use console::Style;
+use serde::Serialize;
+
+use crate::Message;
+
+/// Generic textual message with `ty` prefix.
+///
+/// The type prefix can be stylized in text mode.
+/// e.g. "Tests: 1 passed, 1 failed"
+#[derive(Serialize)]
+pub struct TypedMessage<'a> {
+    ty: &'a str,
+    text: &'a str,
+    #[serde(skip)]
+    type_style: Option<&'a str>,
+}
+
+impl<'a> TypedMessage<'a> {
+    #[must_use]
+    pub fn styled(ty: &'a str, text: &'a str, type_style: &'a str) -> Self {
+        Self {
+            ty,
+            text,
+            type_style: Some(type_style),
```

Comment:
Same here, will we ever allow creating it with it being `None`?

Reply:
Same as https://github.com/foundry-rs/starknet-foundry/pull/3393#discussion_r2106255953